### PR TITLE
[Core] Require correct types for lorisform->addSelect

### PR DESCRIPTION
**Pull request by driusan**
Since the type is currently only enforced by PHPDocs and not
PHP, incorrect types can be passed by addSelect. This enforces
the types from the PHP side of things.

(Currently when a string is passed instead of an array in some
instruments, the array_merge call generates a PHP warning which
breaks endpoints which use the instrument in a JSON context. This
change makes the invalid instruments die with a type error instead,
forcing them to be fixed.)

### CODE REVIEW

File: php/libraries/NDB_Page.class.inc [hunk 1]
```diff
@@ -187,7 +187,7 @@ class NDB_Page implements RequestHandlerInterface
      *
      * @return void
      */
-    function addSelect($name, $label, $options, $attribs=array())
+    function addSelect(string $name, string $label, array $options, array $attribs=array())
```
Comment (johnsaigle) [inline php/libraries/NDB_Page.class.inc:190]: Looks good. Could you types to the arrays in the comment if known? e.g. changing line 183 to

      * @param array<string,string>  $options Options to pass to QuickForm for this

Comment (driusan) [reply php/libraries/NDB_Page.class.inc:190]: I don't know how it's being called, I just based the php types on the phpdoc and know that the places where `$attribs` was a string are wrong..

**Comment by johnsaigle** [comment]
One of the code formatting issues comes from #6733

File: php/libraries/NDB_Page.class.inc (hunk 1 above)
Comment (HenriRabalais) [reply php/libraries/NDB_Page.class.inc:190]: @johnsaigle does php allow type enforcement within arrays now?

Comment (johnsaigle) [reply php/libraries/NDB_Page.class.inc:190]: Not exactly. Some tools know how to analyze the PHPDocs and do additional validation if you specify that all entries need to be a specific type but something else gets passed in. It's not built into PHP itself.

Comment (driusan) [reply php/libraries/NDB_Page.class.inc:190]: @HenriRabalais No, he's referring to the doc comments. phan does analysis based on them in addition to the PHP types

**Pull request merged**